lib/p2p/Peer.ts

```diff
@@ -242,10 +247,14 @@ class Peer extends EventEmitter {
   /**
    * Wait for a packet to be received from peer. Executed on timeout or once packet is received.
    */
-  private wait = (packetType: PacketType) => {
+  private wait = (key: string, timeout?: number) => {
```

Comment:
I'm thinking perhaps `key` go back to `packetType` here. If the `wait` method is always being used to wait for a specific type of packet, as opposed to a response for a packet we've sent, then I think that makes more sense. Also, I'm wondering will we have a use case for `wait` besides the Hello packet? If so, I think I'd prefer explicitly defining `responseMap` as a map between hashes and PendingResponseEntries, and having a separate flag or something to indicate whether we are waiting for a Hello packet.

Reply:
OK, then I think it's good for now but maybe let's revisit this in the future once the p2p layer is fleshed out.